frontend/app/src/stores/menu.ts

```diff
@@ -31,41 +26,40 @@ function close(menu: HTMLElement | undefined): HTMLElement | undefined {
     return undefined;
 }
 
+function positionInCenter(menu: HTMLElement) {
+    const rect = menu.getBoundingClientRect();
+    const dim = centerOfScreen(rect);
+    menu.style.setProperty("left", `${dim.x}px`);
+    menu.style.setProperty("top", `${dim.y + window.scrollY}px`);
+}
+
 export const menuStore = {
     subscribe,
     position: (
-        triggerRect: DOMRect,
+        triggerEl: HTMLElement,
         centered: boolean,
         position: Position = "bottom",
-        align: Alignment = "center",
+        align: Alignment = "middle",
         gutter = 8,
     ): void =>
         update((menu) => {
             if (menu === undefined) return menu;
 
-            const elementRect = menu.getBoundingClientRect();
-            const originalHeight = elementRect.height;
-
-            const dim =
-                centered && get(mobileWidth)
-                    ? centerOfScreen(elementRect)
-                    : boundsCheck(
-                          position,
-                          align,
-                          triggerRect,
-                          derivePosition(triggerRect, elementRect, position, align, gutter),
-                          gutter,
-                      );
-
-            menu.style.setProperty("left", `${dim.x}px`);
-
-            // for landing pages we need to offset based on the window scroll
-            // for the main app this will always be 0 so it's a no-op
-            menu.style.setProperty("top", `${dim.y + window.scrollY}px`);
-            if (originalHeight !== dim.h) {
-                menu.style.setProperty("--override-height", `${dim.h}px`);
+            if (centered && get(mobileWidth)) {
+                positionInCenter(menu);
+            } else {
+                const pos = reposition(triggerEl, menu, {
+                    position: `${position}-${align}`,
+                    margin: gutter,
+                });
+                if (!pos) {
+                    reposition(triggerEl, menu, {
```

Comment:
Couldn't this retry happen within reposition itself? Then also shared by tooltip.

Reply:
yes I guess so. I will change it.